Databases 2 homework

OlhaHumeniuk opened the pull request.

OlhaHumeniuk committed:
Homework

OlhaHumeniuk committed:
Update data

OlhaHumeniuk committed:
Update variables

OlhaHumeniuk committed:
Updating details

OlhaHumeniuk committed:
Adjusting orders

pavlozubariev commented on relational_database/homework.py:
```diff
@@ -19,7 +23,9 @@ def task_1_add_new_record_to_db(con) -> None:
-    pass
+    with con.cursor() as cursor:
+        cursor.execute("INSERT INTO customers(customername,contactname,address,city,postalcode,country) VALUES ('Thomas','David','Some Address','London','774','Singapore');")
+
```
це погана практика хардкодити дані в запитах до бази даних, краще використати плайсхолдери, і нехай psycopg їх заповнить з правильним перетворення типів даних , наприклад cur.execute("INSERT INTO test (num, data) VALUES (%s, %s)",
... (100, "abc'def"))

OlhaHumeniuk replied:
cur.execute("INSERT INTO customers VALUES (%s,%s,%s,%s,%s,%s,%s)",(92,'Thomas','David','Some Address','London','774','Singapore'))
Таке?

OlhaHumeniuk replied:
в  (num, data)  в таком случае, что передавать?

pavlozubariev replied:
(num, data) це приклад якщо потрібно заповнити не всі поля, якщо заповнюєш усі поля до достатньо написати назву таблиці,
маєже правильно має бути cur.execute("INSERT INTO customers VALUES (%s,%s,%s,%s,%s,%s,%s);" % (92,'Thomas','David','Some Address','London','774','Singapore'))
пропустила ; після Values і % перед списком значень щоб працювало форматування стрічки

OlhaHumeniuk replied:
ок ясно, я в статьях не нашла. Может близко где-то у Вас/тебя лежит статья или другой материал по работе с БД в python?

pavlozubariev replied:
вибач варіант з комою був правильний,
cur.execute("INSERT INTO customers VALUES (%s,%s,%s,%s,%s,%s,%s)",(92,'Thomas','David','Some Address','London','774','Singapore'))

pavlozubariev commented on relational_database/homework.py:
```diff
@@ -32,8 +38,8 @@ def task_2_list_all_customers(cur) -> list:
-    pass
-
+    cur.execute("SELECT * FROM customers LIMIT 91;")
+    return cur.fetchall()
```
в базі має бути всього 91 запис тому ліміт не потрібен

OlhaHumeniuk replied:
Удалила.

pavlozubariev commented on relational_database/homework.py:
```diff
@@ -174,7 +190,11 @@ def task_14_list_products_with_supplier_information(cur):
+    FROM products 
+    LEFT JOIN suppliers 
+    ON products.supplierid=suppliers.supplierid;""")
+    return cur.fetchall()
```
неправильно обрані колонки з таблиць, тест не проходить

OlhaHumeniuk replied:
Переделала.

pavlozubariev commented on relational_database/homework.py:
```diff
@@ -198,4 +222,9 @@ def task_16_match_all_customers_and_suppliers_by_country(cur):
+    FULL JOIN suppliers 
+    ON customers.country = suppliers.country
+    ORDER BY customercountry, suppliercountry;""")
+    return cur.fetchall()
```
неправильно обрані колонки з таблиць, тест не проходить

OlhaHumeniuk replied:
Переделала.

OlhaHumeniuk commented:
Thanks for reviewing.

I will check all messages tomorrow morning.

Best regards,

Humeniuk Olha



пн, 30 сент. 2019 г., 21:02 pavlozub85 <notifications@github.com>:

> *@pavlozub85* requested changes on this pull request.
> ------------------------------
>
> In relational_database/homework.py
> <https://github.com/OlhaHumeniuk/common/pull/6#discussion_r329708867>:
>
> > @@ -19,7 +23,9 @@ def task_1_add_new_record_to_db(con) -> None:
>
>      Returns: 92 records
>
>
>
>      """
>
> -    pass
>
> +    with con.cursor() as cursor:
>
> +        cursor.execute("INSERT INTO customers(customername,contactname,address,city,postalcode,country) VALUES ('Thomas','David','Some Address','London','774','Singapore');")
>
> +
>
>
> це погана практика хардкодити дані в запитах до бази даних, краще
> використати плайсхолдери, і нехай psycopg їх заповнить з правильним
> перетворення типів даних , наприклад cur.execute("INSERT INTO test (num,
> data) VALUES (%s, %s)",
> ... (100, "abc'def"))
> ------------------------------
>
> In relational_database/homework.py
> <https://github.com/OlhaHumeniuk/common/pull/6#discussion_r329709217>:
>
> > @@ -32,8 +38,8 @@ def task_2_list_all_customers(cur) -> list:
>
>      Returns: 91 records
>
>
>
>      """
>
> -    pass
>
> -
>
> +    cur.execute("SELECT * FROM customers LIMIT 91;")
>
> +    return cur.fetchall()
>
>
> в базі має бути всього 91 запис тобі ліміт не потрібен
> ------------------------------
>
> In relational_database/homework.py
> <https://github.com/OlhaHumeniuk/common/pull/6#discussion_r329711489>:
>
> > @@ -174,7 +190,11 @@ def task_14_list_products_with_supplier_information(cur):
>
>
>
>      Returns: 77 records
>
>      """
>
> -    pass
>
> +    cur.execute("""SELECT products.productid, products.productname, products.unit, products.price, products.country, products.city, suppliers.suppliername
>
> +    FROM products
>
> +    LEFT JOIN suppliers
>
> +    ON products.supplierid=suppliers.supplierid;""")
>
> +    return cur.fetchall()
>
>
> неправильно обрані колонки з таблиць, тест не проходить
> ------------------------------
>
> In relational_database/homework.py
> <https://github.com/OlhaHumeniuk/common/pull/6#discussion_r329711528>:
>
> > @@ -198,4 +222,9 @@ def task_16_match_all_customers_and_suppliers_by_country(cur):
>
>
>
>      Returns: 194 records
>
>      """
>
> -    pass
>
> +    cur.execute("""SELECT customers.customername, customers.address, customers.country AS customercountry, suppliers.country AS suppliercountry suppliers.suppliername
>
> +    FROM customers
>
> +    FULL JOIN suppliers
>
> +    ON customers.country = suppliers.country
>
> +    ORDER BY customercountry, suppliercountry;""")
>
> +    return cur.fetchall()
>
>
> неправильно обрані колонки з таблиць, тест не проходить
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OlhaHumeniuk/common/pull/6?email_source=notifications&email_token=AMK3RP4YY65XVJD4P6I3AHLQMI5KBA5CNFSM4I3S7X52YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCGLZYBY#pullrequestreview-295148551>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMK3RP5VPH4DHBNJLNPK2TDQMI5KBANCNFSM4I3S7X5Q>
> .
>

OlhaHumeniuk committed:
Delete "LIMIT 91"

OlhaHumeniuk committed:
Updating tasks 14 & 16

should be ok

OlhaHumeniuk committed:
Chaning Insert data function.

("INSERT INTO customers VALUES (%s,%s,%s,%s,%s,%s,%s);" % (92,'Thomas','David','Some Address','London','774','Singapore'))

OlhaHumeniuk committed:
Adjust punctuation
